Ondo-Finance/Role-Changes/src/agent.spec.ts

```diff
@@ -0,0 +1,111 @@
+  GRANT_ROLE = "grantRole",
+  RENOUNCE_ROLE = "renounceRole",
+  REVOKE_ROLE = "revokeRole",
+  NOT_REGISTRY = "notARegistryCall",
+  ENABLE_TOKENS = "enableTokens",
+}
+
+const role = (name: string) => keccak256(name);
+
+const checksumAddress = (prefix: string) =>
+  toChecksumAddress(createAddress(prefix));
+
+const createFinding = (desc: DESC, metadata: Record<string, string>) =>
+  Finding.fromObject({
+    name: "Ondo Finance Registry Role Change Operation",
+    description: `${desc.toString()} call detected`,
+    alertId: "ONDO-1",
+    protocol: "ONDO",
+    type: FindingType.Suspicious,
+    severity: FindingSeverity.Medium,
+    metadata: {
+      ...metadata,
+      method: desc.toString(),
+    },
+  });
+
+describe("Role Changes agent test suite", () => {
+  const registry: string = createAddress("0xdead");
+  const handler: HandleTransaction = agent.provideHandleTransaction(registry);
+
+  it("should return empty findings on empty txns", async () => {
+    const txn: TransactionEvent = new TestTransactionEvent();
+
+    const findings = await handler(txn);
+    expect(findings).toStrictEqual([]);
+  });
+
+  it("should return multiple role changes emitted in the correct registry", async () => {
+    const TEST_CASES: string[][] = [
+      [DESC.ADD_STRATEGIEST.toString(), createAddress("0xe0a30"), createAddress("0x00"), createAddress("0xdef1"), "Strategist-Role-1"],
+      [DESC.GRANT_ROLE.toString()     , createAddress("0xe0a20"), registry, role("Role-1"), createAddress("0xabc01")],
+      [DESC.RENOUNCE_ROLE.toString()  , createAddress("0xe0a10"), createAddress("0x02"), role("Role-2"), createAddress("0xabc02")],
+      [DESC.REVOKE_ROLE.toString()    , createAddress("0xe0a09"), createAddress("0x03"), role("Role-3"), createAddress("0xabc03")],
+      [DESC.ADD_STRATEGIEST.toString(), createAddress("0xe0a08"), registry, createAddress("0xdeade0a"), "Strategist-Role-2"],
+      [DESC.GRANT_ROLE.toString()     , createAddress("0xe0a07"), registry, role("Role-4"), createAddress("0xabc04")],
+      [DESC.RENOUNCE_ROLE.toString()  , createAddress("0xe0a06"), createAddress("0x06"), role("Role-5"), createAddress("0xabc05")],
+      [DESC.REVOKE_ROLE.toString()    , createAddress("0xe0a05"), createAddress("0x07"), role("Role-6"), createAddress("0xabc06")],
+      [DESC.ADD_STRATEGIEST.toString(), createAddress("0xe0a04"), createAddress("0x08"), createAddress("0xfee"), "Strategist-Role-3"],
+      [DESC.GRANT_ROLE.toString()     , createAddress("0xe0a03"), createAddress("0x09"), role("Role-7"), createAddress("0xabc07")],
+      [DESC.RENOUNCE_ROLE.toString()  , createAddress("0xe0a02"), registry, role("Role-8"), createAddress("0xabc08")],
+      [DESC.REVOKE_ROLE.toString()    , createAddress("0xe0a01"), registry, role("Role-9"), createAddress("0xabc10")],
+      [DESC.NOT_REGISTRY.toString()   , createAddress("0xe0a123"), createAddress("0xdef1bad")],
+      [DESC.ENABLE_TOKENS.toString()  , createAddress("0xe0a321"), registry],
+    ];
```

Comment:
I think making an inline comment on which test cases are "wrong"/not supposed to be detected would help a bit with clarity.

Reply:
ok, agree